README.md

```diff
@@ -53,31 +55,56 @@ video = pipe("<my-awesome-prompt>").frames[0]
+Create two files: one containing line-separated prompts, and the other containing line-separated paths to video files (
+video file paths must be relative to the path you specify with `--data_root`). Let's understand this better with an
+example!
 
-Assume you've specified `--data_root` as `/dataset`, and that this directory contains the files: `prompt.txt` and `videos.txt`.
+Assume your specified `--data_root` is `/dataset`, and this directory contains the files `prompts.txt` and `videos.txt`.
 
-The `prompt.txt` file should contain line-separated prompts:
+### Preparing the Prompts Dataset
+
+The `prompts.txt` file should contain line-separated prompts. Please note that the prompts must be in English, and it is
+recommended to use the [prompt polishing script](https://github.com/THUDM/CogVideo/blob/main/inference/convert_demo.py)
+for refinement. Alternatively, you can use [CogVideo-caption](https://huggingface.co/THUDM/cogvlm2-llama3-caption) for
+captioning:
 
 ```
 A black and white animated sequence featuring a rabbit, named Rabbity Ribfried, and an anthropomorphic goat in a musical, playful environment, showcasing their evolving interaction.
-A black and white animated sequence on a ship's deck features a bulldog character, named Bully Bulldoger, showcasing exaggerated facial expressions and body language. The character progresses from confident to focused, then to strained and distressed, displaying a range of emotions as it navigates challenges. The ship's interior remains static in the background, with minimalistic details such as a bell and open door. The character's dynamic movements and changing expressions drive the narrative, with no camera movement to distract from its evolving reactions and physical gestures.
+A black and white animated sequence on a ship’s deck features a bulldog character, named Bully Bulldoger, showcasing exaggerated facial expressions and body language. The character progresses from confident to focused, then to strained and distressed, displaying a range of emotions as it navigates challenges. The ship’s interior remains static in the background, with minimalistic details such as a bell and open door. The character’s dynamic movements and changing expressions drive the narrative, with no camera movement to distract from its evolving reactions and physical gestures.
 ...
+
 ```
 
-The `videos.txt` file should contain line-separate paths to video files. Note that the path should be _relative_ to the `--data_root` directory.
+### Preparing the Video Dataset
+
+The framework supports the following specific resolutions and frame counts:
+
+- **Resolutions (Width * Height) Supported**:
+    - Any width * height divisible by 32 between 256 and 2048. For example, `720 * 480`, `1920 * 1020`.
```

Comment:
This is actually incorrect.

For Text-to-video:
- In 2B model, we use static positional embeddings. So the model needs to initialized with different `sample_height`, `sample_width` and `sample_frames`.
- In 5B model, we use RoPE. So the model can use any resolution pretty much with base resolution as `720 x 480`.

For Image-to-video:
- We only have 5B model. It uses fixed learned positional embeddings so it is quite limited in resolution size. I have noticed that the learned positional embedding values are close to 0 (mean is very close to 0 and variance is extremely small), and even if we remove that layer, generation works quite well. In the future, we could add an option to remove that layer while training and just use RoPE (which can scale to all kinds of resolutions)

Reply:
But I observed that there are actually more restrictions in this part of the code
```
HEIGHT_BUCKETS = [256, 320, 384, 480, 512, 576, 720, 768, 960, 1024, 1280, 1536]
WIDTH_BUCKETS = [256, 320, 384, 480, 512, 576, 720, 768, 960, 1024, 1280, 1536]
FRAME_BUCKETS = [16, 24, 32, 48, 64, 80]
```
In this implementation, the resolution is limited by the three parameters HEIGHT_BUCKETS, WIDTH_BUCKETS, and FRAME_BUCKETS. When processing video, the system will categorize the video's frame count, width, and height into the nearest bucket based on these buckets. If the video's resolution or frame count is not within these buckets, the code will find the bucket closest to the video.
Therefore, if it exceeds 1536, it should be adjusted to 1536; a 1920x1080 video will be adjusted to 1536x1024 for processing. This means the video's height will be adjusted to 1024 and the width to 1536. This is my understanding; should the readme be modified in this way?

Reply:
We could maybe move this to "dataset.md" guide and reference it from the main README?

Reply:
> But I observed that there are actually more restrictions in this part of the code

Those are just the default values if any are not specified by the user actually. We support three different parameters that users should be able to customize: `--height_buckets`, `--width_buckets`, `--frame_buckets`.

So, let's say I wanted to train at 81 x 2048 x 1024 and 49 x 512 x 768 videos, I could run the training with: `--height_buckets 1024 768 --width_buckets 2048 512 --frame_buckets 81 x 49`.

I think if we could better explain this, it would be helpful. But, as such, we are not limited by the frame resolutions that have been hardcoded there

Reply:
> We could maybe move this to "dataset.md" guide and reference it from the main README?我们也许可以把这部分移动到"dataset.md"指南中，并从主 README 中引用它？

I think it is a better idea